src/dfx/src/lib/builders/motoko.rs

```diff
@@ -206,16 +178,13 @@ struct MotokoParams<'a> {
 impl MotokoParams<'_> {
     fn to_args(&self, cmd: &mut std::process::Command) {
         cmd.arg(self.input);
-        cmd.arg("-o").arg(self.output).arg("--idl");
+        cmd.arg("-o").arg(self.output);
         match self.build_target {
             BuildTarget::Release => cmd.args(&["-c", "--release"]),
             BuildTarget::Debug => cmd.args(&["-c", "--debug"]),
-            BuildTarget::Idl => cmd.arg("--idl"),
-            BuildTarget::StableTypes => cmd.arg("--stable-types"),
-        };
-        if matches!(self.build_target, BuildTarget::Release | BuildTarget::Debug) {
-            cmd.arg("--public-metadata").arg("candid:service");
         };
+        cmd.arg("--idl").arg("--stable-types");
```

Comment:
Note that --idl still produce the full idl, including actor class args - will didc check do the sensible thing with that and only look at the actor interface on install, not the class?

Reply:
Yes, didc will skip the init args.